pkg/channel/event_receiver.go

```diff
@@ -192,3 +192,19 @@ func ParseChannel(host url.URL) (ChannelReference, error) {
 		Namespace: chunks[1],
 	}, nil
 }
+
+func translateHostStringToURL(host string) (*url.URL, error) {
+	url, err:= url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// If a scheme is provided (including just "//"), Parse will work as expected.
+	if (url.Host != ""){
+		return url, nil
+	}
+
+	// Otherwise, we prepend a "//" to ensure the provided host is parsed properly.
+	hostWithPrefix := "//" + host
+	return url.Parse(hostWithPrefix)
```

Comment:
This double parse worries me in the hot path. I recommend a different approach. [`TransportContext`](https://github.com/cloudevents/sdk-go/blob/e1f5919ca32388a76b30a230facea51af3dd76c6/pkg/cloudevents/transport/http/context.go#L13-L18) is sadly undocumented but looking at the code I see that tctx.Host is the value of req.Host which will always be an unadorned host string as described in https://golang.org/pkg/net/http/#Request.

If we assume that, we don't need to parse at all: just create a url.URL with the Host field set inline in ServeHTTP.

Maybe TransportContext will provide a url.URL field in the future.